### PR TITLE
fix: allow user to delete default bruno headers in pre-request

### PR DESCRIPTION
**Pull request by shubh-bruno**
### Description

[JIRA](https://usebruno.atlassian.net/browse/BRU-2460)

`req.deleteHeader()` and `req.deleteHeaders()` now correctly suppress default headers (`user-agent`, `accept`, `accept-encoding`, `connection`, etc.) added by Bruno's runtime when used in pre-request scripts.

#### What Was Broken

Calling `req.deleteHeader('user-agent')` in a pre-request script had no effect — the header still arrived at the backend. Same for `accept-encoding`, `connection`, and others.

#### Root Causes & Fixes

| # | Problem | Fix |
|---|---------|-----|
| 1 | `config.headers.delete()` removed the key entirely, so the axios adapter re-added its default (`axios/1.x.x`, `gzip, compress, deflate, br`) via a "don't overwrite if already set" guard | Use `config.headers.set(name, null)` — keeps the key present (blocks the guard) while `toJSON()` omits `null` values from the wire |
| 2 | `connection: keep-alive` is injected by `http.Agent` at the socket level, not by axios — deleting from headers had no effect | Override the agent with `keepAlive: false` per request; in Electron, apply this **after** `setupProxyAgents()` to avoid being overwritten |
| 3 | `deleteHeaders([...])` (plural) never wrote to `req.__headersToDelete`, so it couldn't suppress default headers | Delegate each item to `deleteHeader()` |
| 4 | Network Console showed deleted headers because the timeline was logged before the deletion ran | Move timeline header logging to after the deletion block; skip `null`/`false` values |

### Supported Headers

| Header | Deletable | Notes |
|--------|-----------|-------|
| `user-agent` | Yes | |
| `accept` | Yes | |
| `accept-encoding` | Yes | |
| `connection` | Yes | Agent override — `Connection` header is fully absent |
| `request-start-time` | Yes | Bruno internal header |
| `host` | No | Required by HTTP; silently ignored |

#### Contribution Checklist:

- [ ] **I've used AI significantly to create this pull request**
- [ ] **The pull request only addresses one issue or adds one feature.**
- [ ] **The pull request does not introduce any breaking changes**
- [ ] **I have added screenshots or gifs to help explain the change if applicable.**
- [ ] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [ ] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

#### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

## Summary by CodeRabbit

* **New Features**
  * Selective header removal for outgoing HTTP requests, with safeguards for connection-related headers and more reliable header handling.
  * Request initialization now uses a clean header baseline to ensure deletions and overrides behave predictably.

* **Tests**
  * Added comprehensive tests covering header deletion, duplicate handling, and edge cases.

### CODE REVIEW

File: packages/bruno-cli/src/utils/axios-instance.js [hunk 1]
```diff
@@ -78,14 +78,44 @@ function makeAxiosInstance({ requestMaxRedirects = 5, disableCookies, followRedi
...
+      const http = require('http');
+      const https = require('https');
+      headersToDelete.forEach((headerName) => {
+        const lower = headerName.toLowerCase();
+        if (lower === 'host') return;
+        if (lower === 'connection') {
+          // connection is set by Node.js http.Agent at socket level, not by axios.
+          // keepAlive:false means Node.js does not inject Connection:keep-alive.
+          config.httpAgent = new http.Agent({ keepAlive: false });
+          config.httpsAgent = new https.Agent({ keepAlive: false });
```
Comment (sid-bruno) [inline packages/bruno-cli/src/utils/axios-instance.js]: If the `connection` header needs to be removed, you remove it, there's no need for the agents to be re-initialised, this will conflict with any existing proxy agents that were added

File: packages/bruno-electron/src/ipc/network/axios-instance.js [hunk 2]
```diff
@@ -195,6 +219,16 @@ function makeAxiosInstance({
...
       });
     }
+
+    // Override the agent AFTER setupProxyAgents so keepAlive:true doesn't win.
+    // With keepAlive:false Node.js does not inject Connection:keep-alive at all.
+    if (deleteConnection) {
+      const http = require('http');
+      const https = require('https');
+      config.httpAgent = new http.Agent({ keepAlive: false });
+      config.httpsAgent = new https.Agent({ keepAlive: false });
```
Comment (sid-bruno) [inline packages/bruno-electron/src/ipc/network/axios-instance.js]: Same as https://github.com/usebruno/bruno/pull/7331/changes#r2876590193

File: packages/bruno-requests/src/network/axios-instance.ts [hunk 3]
```diff
@@ -51,10 +52,35 @@ const makeAxiosInstance = (customRequestConfig?: AxiosRequestConfig) => {
...
+    const headersToDelete = modConfig.__headersToDelete;
+    if (headersToDelete && Array.isArray(headersToDelete)) {
+      headersToDelete.forEach((headerName: string) => {
+        const lower = headerName.toLowerCase();
+        if (lower === 'host') return;
+        if (lower === 'connection') {
+          // connection is set by Node.js http.Agent at socket level, not by axios.
+          // keepAlive:false means Node.js does not inject Connection:keep-alive.
+          modConfig.httpAgent = new http.Agent({ keepAlive: false });
+          modConfig.httpsAgent = new https.Agent({ keepAlive: false });
```
Comment (sid-bruno) [inline packages/bruno-requests/src/network/axios-instance.ts]: same as https://github.com/usebruno/bruno/pull/7331/changes#r2876590193
